File: solutions/lesson_08_polymorphism_equals_toString_pbv_pbr/showcases/polymorphism/pow.js
/**
 * This functions calculates the power
 *
 * @param {Number} base the base of the calculation
 * @param {Number} exponent the exponent of the calculation
 * @returns the result of the calculation
 */
function pow(base, exponent = 2) {
	return base ** exponent;
}

console.log(pow(2)); // expected: 4
console.log(pow(2, 3)); // expected: 8

File: solutions/lesson_08_polymorphism_equals_toString_pbv_pbr/showcases/polymorphism/pow.py
def pow(base, exponent=2):
    '''
    Definition
    ----------
    Calculates the power

    Parameters
    ----------
    base: numeric
        the base of the calculation

    exponent: numeric
        the exponent of the calculation

    Returns
    -------
    the result of the calculation
    '''
    return base ** exponent


print(pow(2))  # expected: 4
print(pow(2, 3))  # expected: 8
